README.md

```diff
@@ -50,11 +50,33 @@ If you wish to extend RESTool's functionality and develop on top of it, please g
 
 <br />
 
-## Configuration
+## Configuration Files
 
-One of the best things about RESTool (and the reason we actually built it) is that you don't need to develop anything. Everything is configurable and may be set simply by editing a configuration file (`config.json`).
+### Static
 
-The `config.json` file should be placed in the root folder of the project, alongside with the `index.html` file.
+One of the best things about RESTool (and the reason we actually built it) is that you don't need to develop anything. Everything can be statically configured by simply by editing a configuration file (`config.json`).
+
+An example static configuration file can be found in `public/config-sample.json`.
+
+The contents should be placed in `public/config.json`.
+
+**NOTE:** If you prefer, you can pre-compile the project, and then place your static config files in the `build` directory.
+
+### Dynamic
+
+Dynamic configuration files have additional features. You can, for example, provide filters to pre-process or re-shape data returned by a rest API before displaying it. Dynamic configuration files can be written in Javascript or Typescript.
+#### Javscript
+
+RESTool supports a dynamic js configuration file. To use it, copy `public/static/js/config-sample.js` to `public/static/js/config.js` and edit the file to your liking.
+
+**NOTE:** If you prefer, you can pre-compile the project, and then place your javascript config file in the `build/static/js` directory.
+<br />
+
+#### TypeScript or Javascript in the project tree
```

Comment:
When does the `tsc` command runs? who and when is going to compile the TypeScript based configuration file?

Reply:
Good question. If people use this option, they will write a typescript function that returns a configuration, and they will then run:

npm run build

This will output a deployable app, overwriting the build directory. tsc runs when it is called by `npm i` -> `react-scripts start`.

My initial thought was that I would rather use RESTool as a library, and I still intend to write code to do that. This, however, is a fast interim solution that works for me, and I thought it might be useful for the community in the meantime.

Reply:
I definitely agree that this solution would be great for other developers too, but we need to fine tune it.

A few questions we need to answer:
* Why do we need the export of the ts file to be function?
* If it's a function, let's make it an async one so we could make requests that will affect the config.
* Maybe we should support both function and simply a json export?
* How does the config works in `dev` mode?
* For production, some are using RESTool directly as a module from CDN. I guess it won't work that way?